tests/test_download_live_photos.py

```diff
@@ -0,0 +1,195 @@
+                self._caplog.text,
+            )
+            self.assertIn(
+                f"INFO     Downloading {os.path.join(base_dir, os.path.normpath('2020/11/04/IMG_0514.HEIC'))}",
+                self._caplog.text,
+            )
+            self.assertIn(
+                f"INFO     {os.path.join(base_dir, os.path.normpath('2020/11/04/IMG_0516.HEIC'))} already exists.",
+                self._caplog.text,
+            )
+            self.assertIn(
+                "INFO     All photos have been downloaded!", self._caplog.text
+            )
+            assert result.exit_code == 0
+
+    def test_skip_existing_live_photo_print_filenames(self):
+        base_dir = os.path.normpath("tests/fixtures/Photos")
+        if os.path.exists(base_dir):
+            shutil.rmtree(base_dir)
+        os.makedirs(base_dir)
+
+        # simulate that some of the expected files are there with corret sizes
+        os.makedirs(os.path.join(base_dir, "2020/11/04"))
+        # one photo and one movie are already there and should be skipped
+        # Create dummies with the correct size
+        with open(os.path.join(base_dir, "2020/11/04/IMG_0516.HEIC"), "a") as f:
+            f.truncate(1651485)
+        with open(os.path.join(base_dir, "2020/11/04/IMG_0514_HEVC.MOV"), "a") as f:
+            f.truncate(3951774)
+
+        with vcr.use_cassette("tests/vcr_cassettes/download_live_photos.yml"):
+            # Pass fixed client ID via environment variable
+            os.environ["CLIENT_ID"] = "DE309E26-942E-11E8-92F5-14109FE0B321"
+            runner = CliRunner()
+            result = runner.invoke(
+                main,
+                [
+                    "--username",
+                    "jdoe@gmail.com",
+                    "--password",
+                    "password1",
+                    "--recent",
+                    "3",
+                    "--no-progress-bar",
+                    "--threads-num",
+                    1,
+                    "--only-print-filenames",
+                    "-d",
+                    base_dir,
+                ],
+            )
+            print_result_exception(result)
+
+            filenames = result.output.splitlines()
+
+            print (filenames)
+
+            self.assertEqual(
+                os.path.join(base_dir, os.path.normpath("2020/11/04/IMG_0514.HEIC")),
+                filenames[0]
+            )
+            self.assertEqual(
+                os.path.join(base_dir, os.path.normpath("2020/11/04/IMG_0512.HEIC")),
+                filenames[1]
+            )
+            self.assertEqual(
+                os.path.join(base_dir, os.path.normpath("2020/11/04/IMG_0512_HEVC.MOV")),
+                filenames[2]
+            )
+
+            # Double check that a mocked file does not get listed again. Its already there!
+            assert "2020/11/04/IMG_0514_HEVC.MOV" not in filenames
```

Comment:
Dedup logic will try to download live photo with `-<size>` suffix, correct? If yes, then this assert will not catch that branch of the logic.

Reply:
I hope I get your remark right. Line 193 is not needed to test this bugfix, as if the live-photo (.mov) would get listet for download again (which was the initial bug but does not happen after the fix, of course) the order of the first entries in the file-list changes and one of the 3 checks before already complains (maybe I should have been more descriptive in the comment in line 192). If you use --live-photo-size the bugfix logic chatches that as well, als the check for existence is right before the bugfix.

So I see no need for changing. Or did I miss something?

Reply:
I am talking about `2020/11/04/IMG_0514_HEVC-3951774.MOV` file that will be still downloaded if `2020/11/04/IMG_0514_HEVC.MOV` exists and `--only-print-filenames` selected. Checking the full output of the processing logic will reveal that fact.

Reply:
Exactly. I mocked the .mov intentionally and expect the photo to be downloaded/listed.

Reply:
You are right, my case is edge